src/support.js

```diff
@@ -158,7 +158,33 @@ function loadWebAssemblyModule(binary) {
       'Infinity': Infinity,
     },
     'global.Math': Math,
-    env: env
+    env: env,
+    'asm2wasm': { // special asm2wasm imports
```

Comment:
how about making this and the other place do `'asm2wasm': asm2wasmImports` and we lift the contents out into a singleton in the global scope?